Classes/Service/MauticAuthorizeService.php

```diff
@@ -265,4 +266,79 @@ protected function translate(string $key): string
         }
         return $this->languageService->sL('LLL:EXT:mautic/Resources/Private/Language/locallang_mod.xlf:' . $key);
     }
+
+    public function validateAccessToken(): bool
+    {
+        if ($this->extensionConfiguration['authorizeMode'] === YamlConfiguration::OAUTH1_AUTHORIZATION_MODE) {
+            if ($this->extensionConfiguration['accessToken'] !== ''
+                && $this->extensionConfiguration['accessTokenSecret'] !== ''
+            ) {
+                return true;
+            }
+
+            return false;
+        }
+
+        if ($this->extensionConfiguration['accessToken'] === ''
+            || $this->extensionConfiguration['refreshToken'] === ''
+        ) {
+            return false;
+        }
+
+        if ($this->extensionConfiguration['expires'] > 0
+            && $this->extensionConfiguration['expires'] > time()
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function accessTokenToBeRefreshed(): bool
+    {
+        //Access token have no expire on OAuth 1
+        if ($this->extensionConfiguration['authorizeMode'] === YamlConfiguration::OAUTH1_AUTHORIZATION_MODE) {
+            return false;
+        }
+
+        if ($this->extensionConfiguration['accessToken'] === ''
+            || $this->extensionConfiguration['refreshToken'] === ''
+        ) {
+            return false;
+        }
+
+        if ($this->extensionConfiguration['expires'] > 0
+            && $this->extensionConfiguration['expires'] < time()
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function refreshAccessToken()
+    {
+        try {
+            if ($this->authorization->validateAccessToken()) {
+                if ($this->authorization->accessTokenUpdated()) {
```

Comment:
You can merge these conditions since there is no `else` block:
`if ($this->authorization->validateAccessToken() && $this->authorization->accessTokenUpdated())`